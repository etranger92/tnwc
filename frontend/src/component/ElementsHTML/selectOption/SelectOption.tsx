/* eslint-disable @typescript-eslint/no-unused-vars */
import React, { useState } from "react";
import { IconContext } from "react-icons";
import { FaAngleDown } from "react-icons/fa";
import "./selectOption.css";
//https://react-icons.github.io/react-icons/icons?name=fa

type TProps = {
  // eslint-disable-next-line @typescript-eslint/ban-types
  //  clickEvent: Function;
};
// eslint-disable-next-line no-unused-vars
const SelectOption: React.FC<TProps> = (props) => {
  const [displayOption, setDisplayOption] = useState(false);
  const [classAngleDown, setClassAngle] = useState("");
  const options = [
    "QTLS",
    "AET",
    "CET",
    "DET",
    "CHILD CARE (LVL 2)",
    "CHILD CARE (LVL 3)",
  ];
  const [value, setValue] = useState(options[0]);
  //We remove the first index[0] otherwise it will appear double in the selection and option.
  options.shift();

  const courseSelected = (e: any) => {
    setDisplayOption(!displayOption);
    //We get the child span and his value.
    const spanValue = e.currentTarget.children[0];
    const actuelValue = spanValue.textContent;
    //We replace the value with the previous one
    spanValue.textContent = value;
    //we display the new value
    setValue(actuelValue);
  };
  //const { clickEvent } = props;
  return (
    <div
      id="select-option"
      onMouseEnter={() => {
        setDisplayOption(true);
        setClassAngle("rotate_angle");
      }}
      onMouseLeave={() => {
        setClassAngle("");
        setDisplayOption(false);
      }}
    >
      <div id="select-option-courses">
        <span> {value}</span>
        <span>
          {" "}
          <IconContext.Provider value={{ className: classAngleDown }}>
            <FaAngleDown />
          </IconContext.Provider>
        </span>
      </div>
      <ul id="option-course-container" data-name="selection">
        {options.map((option, index) => (
          <li
            key={"option" + index}
            className={`
          ${displayOption ? "display_option_courses" : ""}`}
            onClick={courseSelected}
          >
            {" "}
            <span>{option}</span>
          </li>
        ))}
      </ul>
    </div>
  );
};
export default SelectOption;
