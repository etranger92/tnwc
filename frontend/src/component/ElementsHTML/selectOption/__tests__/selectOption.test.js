import * as React from 'react';
import SelectOption from "../SelectOption.tsx"
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("We test the innerHTML of the SelectOption", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SelectOption/>);
  });
  test("Snapshot SelectOption", () => {
    //It works
 expect(toJson(wrapper)).toMatchSnapshot();

});
});
