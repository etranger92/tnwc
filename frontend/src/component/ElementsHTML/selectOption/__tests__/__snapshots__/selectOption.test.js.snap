// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We test the innerHTML of the SelectOption Snapshot SelectOption 1`] = `
<div
  id="select-option"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    id="select-option-courses"
  >
    <span>
       
      QTLS
    </span>
    <span>
       
      <ContextProvider
        value={
          Object {
            "className": "",
          }
        }
      >
        <FaAngleDown />
      </ContextProvider>
    </span>
  </div>
  <ul
    data-name="selection"
    id="option-course-container"
  >
    <li
      className="
          "
      key="option0"
      onClick={[Function]}
    >
       
      <span>
        AET
      </span>
    </li>
    <li
      className="
          "
      key="option1"
      onClick={[Function]}
    >
       
      <span>
        CET
      </span>
    </li>
    <li
      className="
          "
      key="option2"
      onClick={[Function]}
    >
       
      <span>
        DET
      </span>
    </li>
    <li
      className="
          "
      key="option3"
      onClick={[Function]}
    >
       
      <span>
        CHILD CARE (LVL 2)
      </span>
    </li>
    <li
      className="
          "
      key="option4"
      onClick={[Function]}
    >
       
      <span>
        CHILD CARE (LVL 3)
      </span>
    </li>
  </ul>
</div>
`;
